fix: allow for no gmaps_key input, add new parameters, fix .onLoad check

laresbernardo opened the pull request:
- **feat:** Allow for users to use `commuting_zones()` without the need of fetching the coordinates using Google's API.
- **feat:** Enable new parameters: quiet, longitude_col_name, latitude_col_name, …
- **fix:** .onLoad warning when checking the code with `devtools::check()`

laresbernardo committed:
fix: allow for no gmaps_key input

laresbernardo committed:
feat: enable parameters quiet, longitude_col_name, latitude_col_name, ... + fix docs format (v0.1.2.9000)

gufengzhou commented:
thanks a thousand:)  while you're doing it already, can you also adapt the readme and the website (under website/doc/Walkthrough/Walkthrough.md), then I can deploy the website change too.

laresbernardo committed:
fix: .onLoad warning message on devtools::check()

laresbernardo committed:
docs: add comment on gmaps_key md file

laresbernardo committed:
docs: add comment on gmaps_key README file

gufengzhou committed:
fix: remove redundant check, quiet message, apply styler & update doc

- remove the redundant ggmap::has_google_key() test, because even a wrong key will return TRUE. For a wrong key, the execution will be stopped with error message when the API test ggmap::geocode("US") fails.
- use the check_gmaps_key alone to decide API key condition
- apply quiet to all messages
- styler:::style_active_pkg()
- updated documentation

gufengzhou committed:
fix: disable plotting when no coordinates available

TSIA

laresbernardo commented:
LGTM!

gufengzhou commented:
Currently testing the integration in GL, because it throws error when running: 
```
GeoTestData_PreTest <- GeoDataRead(data = GeoLift_PreTest,
                                   date_id = "date",
                                   location_id = "location",
                                   Y_id = "Y",
                                   X = c(), #empty list as we have no covariates
                                   format = "yyyy-mm-dd",
                                   cluster_locations =T,
                                   find_location_lat_long = T,
                                   summary = TRUE,
                                   country_name = "United States"
                                   )
```
![image](https://github.com/facebookincubator/CommutingZones/assets/14415136/feac0e4b-adc0-4dcd-82ae-033e76b02395)

FYI @ArturoEsquerra @NicolasMatrices-v2 @JussanN @raphaeltamaki

NicolasMatrices-v2 commented on R/location_cluster_match.R:
```diff
@@ -124,32 +131,36 @@ filter_cluster_file <- function(
+    )
+    if (!quiet) post_msgs(matched_spdf)
+  } else {
+    matched_spdf <- NULL
```
Please replace this line by the following:
```
  stop(
      paste0(
        "Longitude column under the name '", 
        longitude_col_name, 
        "' or latitude column under the name '", 
        latitude_col_name, 
        "' were not found in the location dataset.",
        " Consider manually adding coordinates to locations or including your",
        " GMaps API key by editing the `gmaps_key` parameter."
      )
    )
```
We do not want this function to return something if there Gmaps API is not being used and the lat and long cols are not provided. We want it to fail so functions that use this downstream will also fail.

NicolasMatrices-v2 commented on R/find_lat_long.R:
```diff
@@ -23,92 +25,99 @@
+      warning = function(w) {
+        message(
+          paste0(
+            gsub("\'US\'", "", w$message),
+            "Get the key here https://developers.google.com/maps/documentation/maps-static/get-api-key/"
+          )
+        )
+      },
+      error = function(e) {
+        message(e)
+      }
+    )
+    if (!is.null(test_api_query)) {
+      return(TRUE)
+    }
+  } else {
+    if (!quiet) {
       message(
-        paste0(gsub("\'US\'", "", w$message))
+        paste0(
+          "API key not provided in `gmaps_key`. ",
+          "Proceeding without the coordinates."
+        )
       )
-    },
-    error = function(e) {
-      message(e)
     }
-  )
-
-  if (is.null(test_api_query)) {
-    stop(
-      paste0(
-        "Hint: See `ggmap::register_google()` or provide a valid key",
-        " in the `gmaps_key` parameter."
-      )
-    )
+    return(FALSE)
   }
 }
```
@laresbernardo I was reviewing this check. I edited to allow for the following:
- if users already have their API key registered, then no need to input it in the function (that's how GL works).
- Edited the message when users don't have their APi key registered or it fails.
- Edited the message when users have not provided their API key nor is it registered.

Could you please replace this function for this?
```
check_gmaps_key <- function(gmaps_key = NULL, quiet = FALSE) {
  # Test the provided key
  if (isTRUE(nchar(gmaps_key) > 0)) {
    # Register API Key
    if (!quiet){
      message("Registering Gmaps API key.")
    }
    ggmap::register_google(key = gmaps_key)
  }
  
  if (!ggmap::has_google_key()) {
    if (!quiet) {
      message(
        paste0(
          "API key not provided in `gmaps_key` nor was it previously registered.",
          "\nCoordinates for locations will not be filled in using the GMaps API."
        )
      )
    }
    return(FALSE)
  } else {
    if (!quiet & is.null(gmaps_key)){
      message("Gmaps API key is already registered.")
    }
  }

  # Verify if the provided API key is valid.
  test_api_query <- NULL
  tryCatch(
    {
      test_api_query <- suppressMessages(ggmap::geocode("US"))
    },
    warning = function(w) {
      message(
        paste0(
          gsub("\'US\'", "", w$message),
          "Get the key here https://developers.google.com/maps/documentation/maps-static/get-api-key/"
        )
      )
    },
    error = function(e) {
      message(e)
    }
  )
  if (!is.null(test_api_query)) {
    return(TRUE)
  } else {
    if (!quiet) {
      message(
        "To use this method, you need to register a valid GMaps API key.\n"
      )
    }
    return(FALSE)
  }
}
```
Please let me know if you have comments or questions.

NicolasMatrices-v2 reviewed (changes requested):
Bernie, could you please check the comments on made on the PR before merging? I think there are a couple of things we should edit before landing.

NicolasMatrices-v2 commented on R/location_cluster_match.R:
```diff
@@ -26,7 +26,7 @@
 #'   location_col_name = "location",
 #'   country_col_name = "country",
-#'   gmaps_key = "Enter Your Google API Key Here"
+#'   gmaps_key = "Enter Your Google API Key Here" # Not required
```
Please replace this for this: "#'   gmaps_key = NULL # Not required". thanks

laresbernardo committed:
fix: return error instead of NULL when no coordinates found for location_to_cluster_match()

laresbernardo commented:
@NicolasMatrices-v2 please check. Thanks for your comments and feedback. I tend to avoid `stop()` errors when not 100% necessary; this might be an example where that's the case.

laresbernardo commented:
@NicolasMatrices-v2 @gufengzhou any updates here? Are we ready to merge after the changes suggested?
